transport-native-epoll/src/main/java/io/netty/channel/unix/FileDescriptor.java

```diff
@@ -220,18 +220,12 @@ static boolean isOutputShutdown(int state) {
         return (state & STATE_OUTPUT_SHUTDOWN_MASK) != 0;
     }
 
-    static boolean shouldAttemptShutdown(int state, boolean read, boolean write) {
-        return !isClosed(state) && (read && !isInputShutdown(state) || write && !isOutputShutdown(state));
+    static int inputShutdown(int state) {
+        return state | STATE_INPUT_SHUTDOWN_MASK;
     }
 
-    static int calculateShutdownState(int state, boolean read, boolean write) {
-        if (read) {
-            state |= STATE_INPUT_SHUTDOWN_MASK;
-        }
-        if (write) {
-            state |= STATE_OUTPUT_SHUTDOWN_MASK;
-        }
-        return state;
+    static int outputShutdown(int state) {
+        return state | STATE_OUTPUT_SHUTDOWN_MASK;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)